xilem_web/src/events.rs

```diff
@@ -508,3 +507,109 @@ event_definitions!(
     (OnWaiting, "waiting", Event),
     (OnWheel, "wheel", WheelEvent)
 );
+
+pub struct OnResize<V, State, Action, Callback> {
+    pub(crate) element: V,
+    pub(crate) handler: Callback,
+    pub(crate) phantom_event_ty: PhantomData<fn() -> (State, Action)>,
+}
+
+pub struct OnResizeState<VState> {
+    child_state: VState,
+    // Closures are retained so they can be called by environment
+    #[allow(unused)]
+    callback: Closure<dyn FnMut(js_sys::Array)>,
+    observer: web_sys::ResizeObserver,
+}
+
+impl<V, State, Action, Callback> ViewMarker for OnResize<V, State, Action, Callback> {}
+impl<State, Action, OA, Callback, V: View<State, Action, ViewCtx, DynMessage>>
+    View<State, Action, ViewCtx, DynMessage> for OnResize<V, State, Action, Callback>
+where
+    State: 'static,
+    Action: 'static,
+    OA: OptionalAction<Action>,
+    Callback: Fn(&mut State, web_sys::ResizeObserverEntry) -> OA + 'static,
+    V: View<State, Action, ViewCtx, DynMessage>,
+    V::Element: ElementAsRef<web_sys::Element>,
+{
+    type Element = V::Element;
+
+    type ViewState = OnResizeState<V::ViewState>;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        ctx.with_id(ViewId::new(0), |ctx| {
```

Comment:
I wonder if it would be more idiomatic to use a different number here.

e.g. something like.
```rust
const ON_RESIZE_VIEW_ID: ViewId = ViewId::new(0x1234_ABCD);
```

Which might be slightly more likely to catch a mistake in a different view.

Reply:
Yeah makes sense, needed a `const` change in xilem_core. I applied that for all the event views now.

Reply:
And good, that I just tested this quickly before merging... I introduced a small bug (which is now fixed)...